redfish/resource_redfish_virtual_media.go

```diff
@@ -197,13 +175,70 @@ func createRedfishVirtualMedia(service *gofish.Service, d *schema.ResourceData)
 		WriteProtected:       writeProtected,
 	}
 
-	err = virtualMedia.InsertMediaConfig(virtualMediaConfig)
+	//Get Systems details
+	systems, err := service.Systems()
 	if err != nil {
-		return diag.Errorf("Couldn't mount Virtual Media: %s", err)
+		return diag.Errorf("Error when retrieving systems: %s", err)
 	}
 
-	d.SetId(virtualMedia.ODataID)
-	return diags
+	virtualMediaCollection, err := systems[0].VirtualMedia()
```

Comment:
Please include a check to see if zero systems have been returned by API.

Reply:
added.